### PR TITLE
Adds legacy Java workflow inheritance content to dev guide

### PR DESCRIPTION
**Pull request by flossypurse**
The content in this PR existed in the legacy Java docs and was missed during the migration.
This PR adds it to the Foundations section of the Java dev guide with minimal updates.

### CODE REVIEW

File: assembly/guide-configs/app-dev/java/foundations.json [hunk 1]
```diff
@@ -47,6 +47,10 @@
       "type": "h2",
       "id": "java/how-to-develop-a-workflow-definition-in-java"
     },
+    {
+      "type": "h3",
+      "id": "java/interface-heirarchies"
```
Comment (tomwheeler) [inline assembly/guide-configs/app-dev/java/foundations.json:52]: Probably a legacy issue, but I noticed that "hierarchies" is misspelled here (and in the name of the file itself as well as other references to it).

File: docs-src/java/interface-heirarchies.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+@WorkflowInterface
+public interface Workflow1 extends BaseWorkflow {}
+
+@WorkflowInterface
+public interface Workflow2 extends BaseWorkflow {}
+```
+
+If we attempt to register implementations of Workflow1 and Workflow2 with a Worker will fail.
```
Comment (tomwheeler) [inline docs-src/java/interface-heirarchies.md:78]: There's a grammatical error here that makes the sentence hard to parse. I can interpret the intent in two different ways and it's not clear which is correct, so I am not sure which to suggest. Maybe rephrase as "Any attempt to register implementations for both interfaces with a Worker will fail."

**Review by tomwheeler — CHANGES_REQUESTED** [review]
I would recommend fixing the grammatical issue mentioned in `docs-src/java/interface-heirarchies.md` before publishing this. 

The filename misspelling is less important since the references are at least consistent with one another. It would be great if that could be fixed too, but certainly not essential.

**Review by rachfop — APPROVED** [review]

File: assembly/guide-configs/app-dev/java/foundations.json (hunk 1 above)
Comment (flossypurse) [reply assembly/guide-configs/app-dev/java/foundations.json:52]: No, that's me :) Not sure what I was thinking there.

File: docs-src/java/interface-heirarchies.md (hunk 2 above)
Comment (flossypurse) [reply docs-src/java/interface-heirarchies.md:78]: Good catch.

**Review by tomwheeler — APPROVED** [review]
Ship it!
